src/client/item_visuals_manager.cpp

```diff
@@ -10,10 +10,28 @@
 #include "itemdef.h"
 #include "inventory.h"
 
-ItemVisualsManager::ItemVisuals::~ItemVisuals() {
-	if (wield_mesh.mesh)
-		wield_mesh.mesh->drop();
-}
+struct ItemVisualsManager::ItemVisuals
+{
+	ItemMesh item_mesh;
+	Palette *palette;
+
+	AnimationInfo inventory_normal;
+	AnimationInfo inventory_overlay;
+
+	ItemVisuals() :
+		palette(nullptr)
+	{}
+
+	~ItemVisuals()
+	{
+		inventory_normal.freeFrames();
```

Comment:
wouldn't it be cleaner if `ItemVisuals` owned the frames ptr separately, so that `AnimationInfo` is always just a reference?